evalml/pipelines/utils.py

```diff
@@ -287,6 +289,88 @@ def _get_pipeline_base_class(problem_type):
         return TimeSeriesMulticlassClassificationPipeline
 
 
+def _make_pipeline_time_series(
+    X,
+    y,
+    estimator,
+    problem_type,
+    parameters=None,
+    sampler_name=None,
+    known_in_advance=None,
+):
+    """Make a pipeline for time series problems.
+
+    If there are known-in-advance features, the pipeline will have two parallel subgraphs.
+    In the first part, the not known_in_advance features will be featurized with a TimeSeriesFeaturizer.
+    The known_in_advance features are treated like a non-time-series features since they don't change with time.
+
+    Args:
+         X (pd.DataFrame): The input data of shape [n_samples, n_features].
+         y (pd.Series): The target data of length [n_samples].
+         estimator (Estimator): Estimator for pipeline.
+         problem_type (ProblemTypes or str): Problem type for pipeline to generate.
+         parameters (dict): Dictionary with component names as keys and dictionary of that component's parameters as values.
+             An empty dictionary or None implies using all default values for component parameters.
+         sampler_name (str): The name of the sampler component to add to the pipeline. Only used in classification problems.
+             Defaults to None
+         known_in_advance (list[str], None): List of features that are known in advance.
+
+    Returns:
+        PipelineBase: TimeSeriesPipeline''
+    """
+    if known_in_advance:
+        not_known_in_advance = [c for c in X.columns if c not in known_in_advance]
+        X_not_known_in_advance = X.ww[not_known_in_advance]
+        X_known_in_advance = X.ww[known_in_advance]
+    else:
+        X_not_known_in_advance = X
+        X_known_in_advance = None
+
+    preprocessing_components = _get_preprocessing_components(
+        X_not_known_in_advance, y, problem_type, estimator, sampler_name
+    )
+
+    if known_in_advance:
+        preprocessing_components = [SelectColumns] + preprocessing_components
+    else:
+        preprocessing_components += [estimator]
+
+    component_graph = PipelineBase._make_component_dict_from_component_list(
+        preprocessing_components
+    )
+    base_class = _get_pipeline_base_class(problem_type)
+    pipeline = base_class(component_graph, parameters=parameters)
+    if X_known_in_advance is not None:
+        # Pre-processing components do not depend on problem type so we
+        # are ok by specifying regression for the known-in-advance sub pipeline
+        # Since we specify the correct problem type for the not known-in-advance pipeline
+        # the label encoder and time series featurizer will be correctly added
```

Comment:
I still don't think I fully understand why `problem_type` can't be passed here as well

Reply:
Great question. We can't specify a time series problem type because then the pipeline for the known-in-advance features will have a time series featurizer which is not what we want. We could convert a time series problem type to a non-time-series problem type (time series binary -> binary) but that's the same as just using regression. Since the not-known-in-advance pipeline will have the "right" components for the problem type, e.g. label encoder, samplers, etc, the overall pipeline will have them too.

Reply:
Elaborating on this in the comment!